vertx-platform/src/main/java/org/vertx/java/platform/impl/cli/Starter.java

```diff
@@ -296,25 +307,16 @@ private void runVerticle(boolean zip, boolean module, String main, Args args) {
       instances = 1;
     }
 
-    String configFile = args.map.get("-conf");
-    JsonObject conf;
-
-    if (configFile != null) {
-      try (Scanner scanner = new Scanner(new File(configFile)).useDelimiter("\\A")){
-        String sconf = scanner.next();
-        try {
-          conf = new JsonObject(sconf);
-        } catch (DecodeException e) {
-          log.error("Configuration file does not contain a valid JSON object");
-          return;
-        }
-      } catch (FileNotFoundException e) {
-        log.error("Config file " + configFile + " does not exist");
-        return;
-      }
-    } else {
-      conf = null;
+    String configurationLoaderClass = args.map.get("-configurationLoader");
+    if (configurationLoaderClass == null) {
+      configurationLoaderClass = JsonFileConfigurationLoader.class.getName();
     }
+    ConfigurationLoader configurationLoader = getConfigurationLoader(configurationLoaderClass);
+    if (configurationLoader == null) {
+      return;
```

Comment:
Should this ever be an acceptable outcome or it this an error that should be handled accordingly?